besu/src/main/java/org/hyperledger/besu/cli/util/CommandLineUtils.java

```diff
@@ -99,19 +100,23 @@ public static void checkMultiOptionDependencies(
   private static String getAffectedOptions(
       final CommandLine commandLine, final List<String> dependentOptionsNames) {
     return commandLine.getCommandSpec().options().stream()
-        .filter(option -> Arrays.stream(option.names()).anyMatch(dependentOptionsNames::contains))
         .filter(
-            option -> {
-              try {
-                return !option.stringValues().isEmpty()
-                    || commandLine.getDefaultValueProvider().defaultValue(option) != null;
-              } catch (Exception e) {
-                return false;
-              }
-            })
+            option ->
+                Arrays.stream(option.names()).anyMatch(dependentOptionsNames::contains)
+                    && isOptionSet(option))
         .map(option -> option.names()[0])
         .collect(
             Collectors.collectingAndThen(
                 Collectors.toList(), StringUtils.joiningWithLastDelimiter(", ", " and ")));
   }
+
+  private static boolean isOptionSet(final CommandLine.Model.OptionSpec option) {
+    CommandLine commandLine = option.command().commandLine();
```

Comment:
nit: use final keywork with variable names unless they are meant to be assigned another value later on: `final CommandLine commandLine...`. Similarly, `catch(final Exception e) {`